integration-tests/integration/createIncident/save-additional-comment.spec.js

```diff
@@ -0,0 +1,67 @@
+const TasklistPage = require('../../pages/tasklistPage')
+const SubmittedPage = require('../../pages/submittedPage')
+const { StatementStatus } = require('../../../server/config/types')
+
+context('Add comments to statement', () => {
+  const bookingId = 1001
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubLogin')
+    cy.task('stubOffenderDetails', bookingId)
+    cy.task('stubLocations', 'MDI')
+    cy.task('stubOffenders')
+    cy.task('stubLocation', '357591')
+    cy.task('stubUserDetailsRetrieval', 'Mr Zagato')
+    cy.task('stubUserDetailsRetrieval', 'Mrs Jones')
+    cy.task('stubUserDetailsRetrieval', 'Test User')
+  })
+
+  it('A user can select a specific statement, add to it and then return back to statements page', () => {
+    cy.login(bookingId)
+
+    const tasklistPage = TasklistPage.visit(bookingId)
+    tasklistPage.checkNoPartsComplete()
+
+    const newIncidentPage = tasklistPage.startNewForm()
+    newIncidentPage.fillForm()
+
+    const detailsPage = newIncidentPage.save()
+    detailsPage.fillForm()
+
+    const relocationAndInjuriesPage = detailsPage.save()
+    relocationAndInjuriesPage.fillForm()
+
+    const evidencePage = relocationAndInjuriesPage.save()
+    evidencePage.fillForm()
+
+    const checkAnswersPage = evidencePage.save()
+    checkAnswersPage.clickSubmit()
+
+    const submittedPage = SubmittedPage.verifyOnPage()
+
+    const submitStatementPage = submittedPage.continueToStatement()
+    submitStatementPage.lastTrainingMonth().select('March')
+    submitStatementPage.lastTrainingYear().type('2010')
+    submitStatementPage.jobStartYear().type('1999')
+    submitStatementPage.statement().type('This is my statement')
+
+    const confirmStatementPage = submitStatementPage.submit()
+    const statementSubmittedPage = confirmStatementPage.submit()
+    const incidentsPage = statementSubmittedPage.finish()
+    const { reportId } = incidentsPage.getCompleteRow(0)
+    reportId().then(id =>
+      cy.task('getStatementForUser', { reportId: id, status: StatementStatus.SUBMITTED }).then(() => {
+        cy.get('.govuk-button').click()
+        cy.get('[id="additionalComment"]').type('Some new comment 1')
+        cy.get('[data-qa="save-and-continue"]').click()
+        cy.get('[data-qa="statement"]').contains('Some new comment 1')
```

Comment:
It would be better to use data-qa tags and the `row.index` to identify these uniquely - this test reads like it overrides the existing comment/statement.

Also these should live on the page object so the whole test is written at the same level of abstraction:
```
viewStatementPage.comment(1).should('not', 'exist')

viewStatementPage.newComment().type('aaaa')
viewStatementPage.comment(1).contains('aaaa')
viewStatementPage.save()

viewStatementPage.newComment().type('bbbb')
viewStatementPage.comment(2).contains('bbbb')
viewStatementPage.save()
viewStatementPage.newComment().should('be.empty')
```

Reply:
done